src/pages/workspaces/workspace/Data.js

```diff
@@ -563,17 +566,35 @@ const WorkspaceData = _.flow(
     }
   }
 
-  const loadWdsSchema = async () => {
-    if (isFeaturePreviewEnabled('workspace-data-service') && !getConfig().isProd) {
-      try {
-        setWdsSchema([])
-        setWdsSchemaError(undefined)
-        const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(workspaceId)
-        setWdsSchema(wdsSchema)
-      } catch (error) {
-        setWdsSchemaError(error)
+  const loadWdsSchemaAndUrl = async () => {
+    await Ajax(signal).Apps.getV2AppInfo(workspaceId).then(async apps => {
+      // TODO: Update to pull specifically from /api/apps/v2/{workspaceId}/{appName}
+      if (isFeaturePreviewEnabled('workspace-data-service') && !getConfig().isProd) {
+        try {
+          setWdsSchema([])
+          setWdsSchemaError(undefined)
+          const url = getWdsUrl(apps)
+          setWdsProxyUrl(url)
+          const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(url, workspaceId)
+          setWdsSchema(wdsSchema)
+        } catch (error) {
+          setWdsSchemaError(error)
+        }
       }
+    })
+  }
```

Comment:
Two tweaks here: 
1. ensure all the ajax calls are inside the `if` clause, so we don't attempt to list apps if we don't have to
2. add `isAzureWorkspace` to the if clause - this isn't specific to this PR, but it's a good enhancement!

```suggestion
  const loadWdsSchemaAndUrl = async () => {
    if (isFeaturePreviewEnabled('workspace-data-service') && !getConfig().isProd && isAzureWorkspace) {
      await Ajax(signal).Apps.getV2AppInfo(workspaceId).then(async apps => {
        try {
          setWdsSchema([])
          setWdsSchemaError(undefined)
          const url = getWdsUrl(apps)
          setWdsProxyUrl(url)
          const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(url, workspaceId)
          setWdsSchema(wdsSchema)
        } catch (error) {
          setWdsSchemaError(error)
        }
      })
    }
  }
```